### DISCUSSION TITLE
UnstructuredGrid with Trame+Paraview

### DISCUSSION THREAD

**Discussion by vminedev** [Kitware/trame#165] [Q&A] [answered]
Hello.

I am trying to visualize an UnstructuredGrid with trame and run it using Paraview(pvpython).  
To that I have created a vtkUnstructuredGrid and writed into .vtk file then read the file with LegacyVTKReader() method of the simple package.
Is there any other way to create the UnstructuredGrid  using the simple package of paraview without writing and reading file?

Thanks in advance.

**Comment by jourdain** [comment, marked as answer]
No need to write your dataset to disk to make it available in the proxy pipeline infrastructure.

To add any vtk dataset into ParaView, you need to create a trivial producer which will act as a proxy source for your data. Then you need to bind the output of your trivial producer to your vtk dataset. 

The script below illustrate that binding

```python
import vtk
from paraview import simple

cone_source = vtk.vtkConeSource()
cone_source.Update()
dataset = cone_source.GetOutput()

pv_source = simple.TrivialProducer()
pv_source.GetClientSideObject().SetOutput(dataset)

simple.Show()
simple.Render()
```
